src/components/EmojiPicker/EmojiPickerMenu/EmojiPickerMenu.jsx

```diff
@@ -0,0 +1,554 @@
+import {withOnyx} from 'react-native-onyx';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import lodashGet from 'lodash/get';
+import CONST from '../../../CONST';
+import ONYXKEYS from '../../../ONYXKEYS';
+import styles from '../../../styles/styles';
+import * as StyleUtils from '../../../styles/StyleUtils';
+import emojiAssets from '../../../../assets/emojis';
+import EmojiPickerMenuItem from '../EmojiPickerMenuItem';
+import Text from '../../Text';
+import withWindowDimensions, {windowDimensionsPropTypes} from '../../withWindowDimensions';
+import withLocalize, {withLocalizePropTypes} from '../../withLocalize';
+import compose from '../../../libs/compose';
+import getOperatingSystem from '../../../libs/getOperatingSystem';
+import * as User from '../../../libs/actions/User';
+import EmojiSkinToneList from '../EmojiSkinToneList';
+import * as EmojiUtils from '../../../libs/EmojiUtils';
+import CategoryShortcutBar from '../CategoryShortcutBar';
+import TextInput from '../../TextInput';
+import isEnterWhileComposition from '../../../libs/KeyboardShortcut/isEnterWhileComposition';
+import canFocusInputOnScreenFocus from '../../../libs/canFocusInputOnScreenFocus';
+
+const propTypes = {
+    /** Function to add the selected emoji to the main compose text input */
+    onEmojiSelected: PropTypes.func.isRequired,
+
+    /** The ref to the search input (may be null on small screen widths) */
+    forwardedRef: PropTypes.func,
+
+    /** Stores user's preferred skin tone */
+    preferredSkinTone: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    /** Stores user's frequently used emojis */
+    // eslint-disable-next-line react/forbid-prop-types
+    frequentlyUsedEmojis: PropTypes.arrayOf(PropTypes.object),
+
+    /** Props related to the dimensions of the window */
+    ...windowDimensionsPropTypes,
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    forwardedRef: () => {},
+    preferredSkinTone: CONST.EMOJI_DEFAULT_SKIN_TONE,
+    frequentlyUsedEmojis: [],
+};
+
+const EmojiPickerMenu = (props) => {
+    const {forwardedRef, frequentlyUsedEmojis, preferredSkinTone, onEmojiSelected, preferredLocale, isSmallScreenWidth, windowWidth, windowHeight, translate} = props;
+
+    // Ref for the emoji search input
+    const searchInputRef = useRef(null);
+
+    // Ref for emoji FlatList
+    const emojiListRef = useRef(null);
+
+    // We want consistent auto focus behavior on input between native and mWeb so we have some auto focus management code that will
+    // prevent auto focus when open picker for mobile device
+    const shouldFocusInputOnScreenFocus = canFocusInputOnScreenFocus();
+
+    const firstNonHeaderIndex = useRef(0);
+
+    const emojis = useRef([]);
+    if (emojis.current.length === 0) {
+        emojis.current = getEmojisAndHeaderRowIndices().filteredEmojis;
+    }
+    const headerRowIndices = useRef([]);
```

Comment:
Here, after observation i found that `headerRowIndices` ref serves no special purpose. It just stores the same value that `headerIndices` does.


>For example, Unlike `headerRowIndices` ref, the `emoji` ref has a special purpose: 
>
>Observe on [line 466](https://github.com/Expensify/App/pull/29241/files#diff-2051a9ac34f0c7acb64147b79f93a914bc61ed6cd43758098aea8d5119bbd622R466), the `emoji` ref is helping to set the `isFiltered` variable

If you think that my suggestion is valid, I can remove all instaces of `headerRowIndices` ref and I will replace it with the `headerIndices` state instead